src/dune_engine/scheduler.ml

```diff
@@ -1097,3 +1223,27 @@ let inject_memo_invalidation invalidation =
   let* t = t () in
   Event.Queue.send_invalidation_event t.events invalidation;
   Fiber.return ()
+
+let wait_for_process_with_timeout t pid ~timeout =
```

Comment:
By the way, this is not a great way to "timeout" a process because it kills one process instead of killing a process group. The "timeout" command-line tool creates a process group and kills the entire group, which is better if the command you're running is going to have child processes.

I suggest adding that to a comment and to take that into account when writing tests.

Reply:
Good point. In fact, isn't this a problem with our process clean up in general? We could writing bindings for `killpg` and address this in both places.

Reply:
Indeed. Now that you say it I think we've an issue open about it somewhere.

Reply:
By the way, it's not simply a matter of writing bindings, it also means that we'd have to spawn every process in its own process group, which means that we need a patch for spawn and that C-c will no longer be sent to them etc. All of that is probably fine, just not such a self-contained change as one might expect.